### ISSUE TITLE
[IntuneDeviceConfigurationPolicyWindows10] NetworkProxyServer property incorrectly handled

### ISSUE THREAD

**Issue by ykuijs** [microsoft/Microsoft365DSC#2867]
#### Details of the scenario you tried and the problem that is occurring
The NetworkProxyServer property is defined as a string array in the schema of the resource:
https://github.com/microsoft/Microsoft365DSC/blob/296250b4bb8bced33f8c67f6d056b8471911e4c1/Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyWindows10/MSFT_IntuneDeviceConfigurationPolicyWindows10.schema.mof#L162

However, the Get method returns the property directly and is outputting a Dictionary object. In the Export, this is resulting in the following output:
```PowerShell
NetworkProxyServer = System.Collections.Generic.Dictionary`2[System.String,System.Object];
```

When reviewing the NetworkProxyServer property, the output is this:
```
Key                  Value                           
---                  -----                           
address              proxy.contoso.com:8080          
exceptions           {*.contoso.com...               
useForLocalAddresses False 
```

#### Verbose logs showing the problem
```
##[error]Property networkProxyServer in payload has a value that does not match schema.
    + CategoryInfo          : InvalidOperation: ({ body = Micros...onfiguration1 }:) [], CimException
    + FullyQualifiedErrorId : BadRequest,Microsoft.Graph.PowerShell.Cmdlets.NewMgDeviceManagementDeviceConfiguration_C 
   reateExpanded
    + PSComputerName        : localhost
```

#### Suggested solution to the issue
I think the fix is to update the NetworkProxyServer property to the a CIMInstance:
Address: String
Exceptions: String[]
UseForLocalAddresses: Boolean

#### The operating system the target node is running
Windows 11

#### Version of the DSC module that was used ('dev' if using current dev branch)
dev

**Comment by ykuijs** [comment]
The example also needs to get updated to reflect the change.

**Comment by William-Francillette** [comment]
We probably should generate this resource with the DRG to make sure all embedded CIM are used for complex parameters
It will be a breaking change

**Comment by ykuijs** [comment]
Closed via #3118
